### PR TITLE
Add aten::_weight_norm_interface and aten::_weight_norm_interface_backward

### PR DESCRIPTION
**Pull request by huaiyuzh**
add aten::_weight_norm_interface and aten::_weight_norm_interface_backward

### CODE REVIEW

File: src/ATen/native/xpu/WeightNorm.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#include <ATen/native/xpu/sycl/WeightNormKernels.h>
+#include <ATen/xpu/XPUNativeFunctions.h>
+namespace at {
+std::tuple<Tensor, Tensor> XPUNativeFunctions::_weight_norm_interface(
+    const Tensor& v,
+    const Tensor& g,
+    int64_t dim) {
+  return native::xpu::_weight_norm_interface_kernel(v, g, dim);
```
Comment (fengyuan14) [inline src/ATen/native/xpu/WeightNorm.cpp:8]: `_` is meaningless for us, but meaningful for ATen. It means the operator is a low level / internal operator. Please remove it.
`weight_norm_kernel` is good to me.

File: src/ATen/native/xpu/WeightNorm.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    const Tensor& saved_norms,
+    int64_t dim) {
+  TORCH_CHECK(saved_v.is_contiguous(), "saved_v must be contiguous");
+  TORCH_CHECK(saved_g.is_contiguous(), "saved_g must be contiguous");
+  TORCH_CHECK(saved_norms.is_contiguous(), "saved_norms must be contiguous");
+  TORCH_CHECK(
+      dim == 0 || dim == saved_v.dim() - 1,
+      "fused kernels can only be applied for first or last dim")
+
+  return native::xpu::_weight_norm_interface_backward_kernel(
```
Comment (fengyuan14) [inline src/ATen/native/xpu/WeightNorm.cpp:24]: Ditto. `weight_norm_backward_kernel`.

File: src/ATen/native/xpu/sycl/BatchKernel.h [hunk 3]
```diff
@@ -52,11 +96,13 @@ class BatchKernelConfig {
 
   template <class KernelClass>
   void build() {
-    size_t wg_size = syclMaxWorkGroupSize<KernelClass>();
+    size_t wg_size = 0;
```
Comment (fengyuan14) [inline src/ATen/native/xpu/sycl/BatchKernel.h:99]:
```suggestion
    size_t wg_size;
```

File: src/ATen/native/xpu/sycl/BatchKernel.h [hunk 4]
```diff
@@ -52,11 +96,13 @@ class BatchKernelConfig {
...
     size_t sg_size = syclMaxSubGroupSize();
     if (prefer_wg_size_ != 0 && prefer_wg_size_ % sg_size == 0 &&
```
Comment (fengyuan14) [inline src/ATen/native/xpu/sycl/BatchKernel.h:101]:
```suggestion
    // Caller takes responsibility of if work group size is valid or compatible.
    if (prefer_wg_size_ != 0 && prefer_wg_size_ % sg_size == 0 &&
```

File: src/ATen/native/xpu/sycl/BatchKernel.h [hunk 5]
```diff
@@ -247,12 +293,17 @@ class BatchKernelConfig {
       return Policy::pAdaptive;
     }
 
-    BatchKernelConfig cfg_ = {
-        batch, problem, stride, batch * stride, problem_along_x, Policy::pLoop};
+    BatchKernelConfig cfg_ = make_config(
```
Comment (fengyuan14) [inline src/ATen/native/xpu/sycl/BatchKernel.h:296]:
```suggestion
    // Using device max work group size to deduce range configuration approximately.
    BatchKernelConfig cfg_ = make_config(
```

File: src/ATen/native/xpu/sycl/BatchKernel.h [hunk 6]
```diff
@@ -25,6 +25,50 @@ class BatchKernelConfig {
...
+        problem_batch,
+        problem_along_x,
+        policies,
+        prefer_wg_size};
+    cfg.template build<KernelClass>();
+
+    return cfg;
+  }
+
   BatchKernelConfig(
```
Comment (fengyuan14) [inline src/ATen/native/xpu/sycl/BatchKernel.h:72]: We need ban using constructor? Using private to decorate it?

**Comment by fengyuan14** [comment]
Weight norm is not operator in PyTorch frontend, but a parameter utils to normalize parameters. See cases,
test_nn_xpu.py::TestNN::test_weight_norm PASSED                          [ 70%]
test_nn_xpu.py::TestNN::test_weight_norm_pickle PASSED                   [ 70%]
nn/test_lazy_modules_xpu.py::TestLazyModules::test_weight_norm PASSED    [100%]
nn/test_parametrization_xpu.py::TestNNParametrization::test_weight_norm_deepcopy_swap_False PASSED [ 86%]
nn/test_parametrization_xpu.py::TestNNParametrization::test_weight_norm_deepcopy_swap_True PASSED [ 87%]
nn/test_parametrization_xpu.py::TestNNParametrization::test_weight_norm_pickle_swap_False PASSED [ 89%]
nn/test_parametrization_xpu.py::TestNNParametrization::test_weight_norm_pickle_swap_True PASSED [ 91%]
nn/test_parametrization_xpu.py::TestNNParametrization::test_weight_norm_state_dict_compat_swap_False PASSED [ 93%]
nn/test_parametrization_xpu.py::TestNNParametrization::test_weight_norm_state_dict_compat_swap_True PASSED [ 94%]
nn/test_parametrization_xpu.py::TestNNParametrizationDeviceXPU::test_weight_norm_parametrization_swap_False_xpu PASSED [ 98%]
nn/test_parametrization_xpu.py::TestNNParametrizationDeviceXPU::test_weight_norm_parametrization_swap_True_xpu PASSED [100%]

**Review by fengyuan14 — APPROVED** [review]
